Mathlib/Data/Bitvec/Defs.lean

```diff
@@ -1,339 +1,141 @@
-/-- bitwise not -/
-protected def not (bv : Bitvec n) : Bitvec n :=
-  map not bv
-#align bitvec.not Bitvec.not
-
--- porting note: added protected, since now this clashes with `_root_.and` (formerly `band`)
-/-- bitwise and -/
-protected def and : Bitvec n → Bitvec n → Bitvec n :=
-  map₂ and
-#align bitvec.and Bitvec.and
-
--- porting note: added protected, since now this clashes with `_root_.or` (formerly `bor`)
-/-- bitwise or -/
-protected def or : Bitvec n → Bitvec n → Bitvec n :=
-  map₂ or
-#align bitvec.or Bitvec.or
-
--- porting note: added protected, since now this clashes with `_root_.xor` (formerly `bxor`)
-/-- bitwise xor -/
-protected def xor : Bitvec n → Bitvec n → Bitvec n :=
-  map₂ xor
-#align bitvec.xor Bitvec.xor
-
-instance : Complement (Bitvec n) :=
-⟨Bitvec.not⟩
-
-instance : AndOp (Bitvec n) :=
-⟨Bitvec.and⟩
-
-instance : OrOp (Bitvec n) :=
-⟨Bitvec.or⟩
-
-instance : Xor (Bitvec n) :=
-⟨Bitvec.xor⟩
-
-end Bitwise
+#align bitvec.not Std.BitVec.not
+#align bitvec.and Std.BitVec.and
+#align bitvec.or Std.BitVec.or
+#align bitvec.xor Std.BitVec.xor
 
 /-! ### Arithmetic operators -/
 
-
-section Arith
-
-variable {n : ℕ}
-
-/-- `neg x` is the two's complement of `x`. -/
-protected def neg (x : Bitvec n) : Bitvec n :=
-  let f y c := (y || c, xor y c)
-  Prod.snd (mapAccumr f x false)
-#align bitvec.neg Bitvec.neg
-
+#align bitvec.neg Std.BitVec.neg
 /-- Add with carry (no overflow) -/
-def adc (x y : Bitvec n) (c : Bool) : Bitvec (n + 1) :=
-  let f x y c := (Bool.carry x y c, Bool.xor3 x y c)
-  let ⟨c, z⟩ := Vector.mapAccumr₂ f x y c
-  c ::ᵥ z
-#align bitvec.adc Bitvec.adc
+def adc {n} (x y : BitVec n) (c : Bool) : BitVec (n+1) :=
+  ofFin (x.toNat + y.toNat + c.toNat)
+#align bitvec.adc Std.BitVec.adc
 
-/-- The sum of two bitvectors -/
-protected def add (x y : Bitvec n) : Bitvec n :=
-  tail (adc x y false)
-#align bitvec.add Bitvec.add
+#align bitvec.add Std.BitVec.add
 
 /-- Subtract with borrow -/
-def sbb (x y : Bitvec n) (b : Bool) : Bool × Bitvec n :=
-  let f x y c := (Bool.carry (not x) y c, Bool.xor3 x y c)
-  Vector.mapAccumr₂ f x y b
-#align bitvec.sbb Bitvec.sbb
+def sbb {n} (x y : BitVec n) (b : Bool) : Bool × BitVec n :=
+  let y := y + ofFin b.toNat
+  (x < y, x - y)
+#align bitvec.sbb Std.BitVec.sbb
 
-/-- The difference of two bitvectors -/
-protected def sub (x y : Bitvec n) : Bitvec n :=
-  Prod.snd (sbb x y false)
-#align bitvec.sub Bitvec.sub
+#align bitvec.sub Std.BitVec.sub
+#align bitvec.mul Std.BitVec.mul
 
-instance : Zero (Bitvec n) :=
-  ⟨Bitvec.zero n⟩
-
-instance : One (Bitvec n) :=
-  ⟨Bitvec.one n⟩
+/-! ### Comparison operators -/
 
-instance : Add (Bitvec n) :=
-  ⟨Bitvec.add⟩
+#align bitvec.uborrow Std.BitVec.ult
+#align bitvec.ult Std.BitVec.ult
 
-instance : Sub (Bitvec n) :=
-  ⟨Bitvec.sub⟩
+/-- Unsigned greater than for bitvectors. -/
+protected def ugt (x y : BitVec w) : Bool := BitVec.ult y x
+#align bitvec.ugt Std.BitVec.ugt
```

Comment:
What's the rationale for changing this and the declarations around it from `Prop` to `bool`? Can you add an explanation in the PR description?

Reply:
The choice was made to be consistent with Std, I believe also with input from an earlier review by @ChrisHughes24. I've edited the PR description to clarify the motivation behind this change